### PR TITLE
fix(timepicker): Manual keyboard input broken (#143)

### PR DESCRIPTION
**Pull request by RistiCore**

### CODE REVIEW

**Comment by lskramarov** [comment]
Нужно покрыть этот кейс тестами.

**Comment by RistiCore** [comment]
> Нужно покрыть этот кейс тестами.

Написан тест.

File: packages/mosaic/timepicker/timepicker.spec.ts [hunk 1]
```diff
@@ -594,5 +596,37 @@ describe('McTimepicker', () => {
...
+                    inputElementDebug.nativeElement.selectionEnd = 1;
+                    dispatchFakeEvent(inputElementDebug.nativeElement, 'input');
+                    fixture.detectChanges();
+
+                    expect(inputElementDebug.nativeElement.value).toBe('1:00', 'Keypress works!');
+
+                    const key2PressEvent: KeyboardEvent = createKeyboardEvent('keydown', TWO);
+                    dispatchEvent(inputElementDebug.nativeElement, key2PressEvent);
+                    inputElementDebug.nativeElement.value =
+                        `${inputElementDebug.nativeElement.value.substring(0, 1)}` +
```
Comment (lskramarov) [inline packages/mosaic/timepicker/timepicker.spec.ts]: why do you use concatenating template literals ? in this case you should use only template literals.

File: packages/mosaic/timepicker/timepicker.spec.ts [hunk 2]
```diff
@@ -594,5 +596,37 @@ describe('McTimepicker', () => {
                     expect(inputElementDebug.nativeElement.value).toBe('23:59', 'Arrow-Down key decrement not working');
                 });
         });
+
+        it('Manual keyboard input digit-by-digit', () => {
+            return fixture.whenStable()
+                .then(() => {
+                    dispatchFakeEvent(inputElementDebug.nativeElement, 'focus');
```
Comment (lskramarov) [inline packages/mosaic/timepicker/timepicker.spec.ts]: Do you really need dispatch this events ?

File: packages/mosaic/timepicker/timepicker.spec.ts [hunk 3]
```diff
@@ -594,5 +596,37 @@ describe('McTimepicker', () => {
...
+                    inputElementDebug.nativeElement.selectionStart = 0;
```
[the remark this answers is not in the document]
Comment (RistiCore) [reply packages/mosaic/timepicker/timepicker.spec.ts]: Done.

File: packages/mosaic/timepicker/timepicker.spec.ts [hunk 4]
```diff
@@ -594,5 +596,37 @@ describe('McTimepicker', () => {
...
+
+                    const key1PressEvent: KeyboardEvent = createKeyboardEvent('keydown', ONE);
+                    dispatchEvent(inputElementDebug.nativeElement, key1PressEvent);
+                    inputElementDebug.nativeElement.value = `1${inputElementDebug.nativeElement.value.substring(2)}`;
+                    inputElementDebug.nativeElement.selectionStart = 1;
+                    inputElementDebug.nativeElement.selectionEnd = 1;
+                    dispatchFakeEvent(inputElementDebug.nativeElement, 'input');
+                    fixture.detectChanges();
+
+                    expect(inputElementDebug.nativeElement.value).toBe('1:00', 'Keypress works!');
```
[the remark this answers is not in the document]
Comment (RistiCore) [reply packages/mosaic/timepicker/timepicker.spec.ts]: Message fixed

File: packages/mosaic/timepicker/timepicker.spec.ts (hunk 1 above)
Comment (RistiCore) [reply packages/mosaic/timepicker/timepicker.spec.ts]: I need to produce single-line string, but template literal with line breaks produces string with line breaks.

Template literals without line-breaks causes long line (>120 symbols) in source code.

Using simple strings with concatenation causes tslint to warn `prefer template strings`.

File: packages/mosaic/timepicker/timepicker.spec.ts (hunk 2 above)
Comment (RistiCore) [reply packages/mosaic/timepicker/timepicker.spec.ts]: No, removed.

File: packages/mosaic/timepicker/timepicker.spec.ts (hunk 1 above)
Comment (RistiCore) [reply packages/mosaic/timepicker/timepicker.spec.ts]: I've splitted this string on two variables and then used single template string.

File: packages/mosaic/timepicker/timepicker.spec.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (lskramarov) [reply packages/mosaic/timepicker/timepicker.spec.ts]: not fixed yet... you have two expects

[the remark this answers is not in the document]
Comment (RistiCore) [reply packages/mosaic/timepicker/timepicker.spec.ts]: Yep, fixed two ones.

**Pull request merged**